cetmix_tower_server/models/cx_tower_server_template.py

```diff
@@ -356,19 +362,41 @@ def _prepare_server_values(self, **kwargs):
 
     def _validate_required_variables(self, configuration_variables):
         """
-        Validate that all required variables are present and not empty.
+        Validate that all required variables are present, not empty,
+        and that no required variable is entirely missing from the configuration.
 
         Args:
             configuration_variables (dict): A dictionary of variable references
                                              and their values.
 
         Raises:
-            ValidationError: If any required variables are missing or empty.
+            ValidationError: If all required variables are m
+                            issing from the configuration,
+                            or if any required variable is empty or missing.
         """
         missing_variables = []
         empty_variables = []
 
-        for variable in self.variable_value_ids.filtered("required"):
+        # We get a list of required variables
+        required_variables = self.variable_value_ids.filtered("required")
+
+        # Checking if all required variables are missing
+        if required_variables and not any(
```

Comment:
We need to check if all required variables are present. Which can be done using `all` operator. So using  `..not any (..` seems a little bit misleading.
You can also skip this check at all and search for the missing variables right away. 
So you could raise an exception with those found or proceed further if there are none.